pytorch_lightning/utilities/enums.py

```diff
@@ -34,7 +34,7 @@ def __eq__(self, other: Union[str, Enum]) -> bool:
     def __hash__(self) -> int:
         # re-enable hashtable so it can be used as a dict key or in a set
         # example: set(LightningEnum)
-        return hash(self.name)
+        return hash(self.value)
```

Comment:
Need this so it is also case invariant for hash. So this works just as `__eq__`

```suggestion
        return hash(self.value.lower())
```